### ISSUE TITLE
## Issue

### ISSUE THREAD

**Issue by dmcstravick7** [kedacore/keda#4772]
## Issue
- metrics-server APIservice is failing to come up (ServiceNotFound) `v1beta1.external.metrics.k8s.io`
- All Keda resources are deployed as `keda-operator*` in this case that includes `keda-operator-metrics-apiserver`. 
  - The spec for `/keda/config/metrics-server/api_service.yaml` contains spec.Service.Name for `keda-metrics-apiserver`.  **^ Issue**

## Fix
- Editing the APIservice and updating spec.Service.Name to `keda-operator-metrics-apiserver`.
- E.g 
  - ``` kubectl edit apiservice v1beta1.external.metrics.k8s.io ``` 
  - Change  spec.Service.Name from `keda-metrics-apiserver` --> `keda-operator-metrics-apiserver`.

## Steps to reproduce
 - Install Keda via helm chart 2.11.1 (note this has happened with previous versions)
 - Running any kubectl command such as `kubectl get pods` returns the below error (also returns pods)
 ```
57503 memcache.go:287] couldn't get resource list for external.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
 ```

## Notes
- Could this be due to setting me not setting some value in the chart that I am unaware of?

_Originally posted by @dmcstravick7 in https://github.com/kedacore/keda/discussions/4769_

**Comment by tomkerkhove** [comment]
Is this a Helm issue or KEDA deployment itself? Always happy to review PRs.

**Comment by dmcstravick7** [comment]
I think it's a Keda issue as the Kubernetes manifest specifies the name wrong, which is then picked up by Helm in my case.

I'll create a PR, I just want to be very sure that this is an issue for someone other then me.

**Comment by tomkerkhove** [comment]
Yes, but the main question is do we need to update the Helm chart and/or KEDA core which generates the manifests. @zroubalik can you take a look? I keep forgetting where we annotate this.

**Comment by Apurva-Jyoti-paul** [comment]
I am having this same issue for myself too

**Comment by JorTurFer** [comment]
Hello,
I have just tried with a fresh cluster using latest helm chart (v2.11.1) and default values and I can't reproduce the issue. In my case, the apiservice points to the correct service. 

Could you share your values to test them?

@tomkerkhove , this only can happen with helm because otherwise, e2e test wouldn't pass as the metrics server hadn't been reachable, so I move this issue to charts repo
